frontend/src/features/public-form/PublicFormProvider.tsx

```diff
@@ -231,7 +235,9 @@ export const PublicFormProvider = ({
         captchaResponse,
         responseMetadata: {
           responseTimeMs: differenceInMilliseconds(Date.now(), startTime),
-          numVisibleFields,
+          numVisibleFields: isPaymentEnabled
+            ? numVisibleFields + 1
+            : numVisibleFields,
```

Comment:
Jut to note: this is case where javascript automatic type conversion would work well (boolean to int: `true->1`, `false->0`), but it's not recommended for readability of course 😅 

```javascript
          numVisibleFields: numVisibleFields + isPaymentEnabled
```

Reply:
Omg this is real big brain! 

HAHAHA though typescript will complain, and I think we need to typecast 😅

I think I will keep it to the current implementation too cause with payments by product, we might wanna revisit how we calculate payment fields in our responseTime as well (so may not be `1`?)

Thanks @timotheeg!